src/specBuilder/legend/legendSpecBuilder.test.ts

```diff
@@ -355,4 +377,43 @@ describe('addSignals()', () => {
 			)
 		).toBeUndefined();
 	});
+
+	test('should update colorAnimationDirection signal if animations are true', () => {
+		defaultSpec.signals = new Array<Signal>;
+		defaultSpec.signals.push(...defaultAnimationsSignals);
+		expect(addLegend(defaultSpec, { ...defaultLegendProps, animations: true }).signals)
+			.toStrictEqual([
+				{
```

Comment:
We could make this test more readable by extracting the expected object into a constant in the top of the file